include/os/linux/kernel/linux/simd_x86.h

```diff
@@ -136,29 +132,79 @@
  * When the kernel_fpu_* symbols are unavailable then provide our own
  * versions which allow the FPU to be safely used.
  */
-#if defined(HAVE_KERNEL_FPU_INTERNAL) || defined(HAVE_KERNEL_FPU_XSAVE_INTERNAL)
+#if defined(HAVE_KERNEL_FPU_INTERNAL)
 
-#if defined(HAVE_KERNEL_FPU_XSAVE_INTERNAL)
 /*
- * Some sanity checks.
- * HAVE_KERNEL_FPU_INTERNAL and HAVE_KERNEL_FPU_XSAVE_INTERNAL are exclusive.
+ * For kernels not exporting *kfpu_{begin,end} we have to use inline assembly
+ * with the XSAVE{,OPT,S} instructions, so we need the toolchain to support at
+ * least XSAVE.
  */
-#if defined(HAVE_KERNEL_FPU_INTERNAL)
-#error "HAVE_KERNEL_FPU_INTERNAL and HAVE_KERNEL_FPU_XSAVE_INTERNAL defined"
-#endif
 /*
- * For kernels >= 5.16 we have to use inline assembly with the XSAVE{,OPT,S}
- * instructions, so we need the toolchain to support at least XSAVE.
+ * XXXX We could define the opcode with .byte and not use the instruction to
+ * circumvent all #idef HAVE_XSAVE{,OPT,S} ???
```

Comment:
The way things are currently structured strikes me a pretty reasonable.

Reply:
All right, will keep as is.